test/test-async.js

```diff
@@ -1889,8 +1890,9 @@ exports['map'] = {
     var r = [];
     async.map(a, function(x, callback){
         r.push(x);
+        var done = r.length == a.length;
         callback(null);
-        if (r.length >= a.length) {
+        if (done) {
             test.same(r, a);
```

Comment:
Test fail without them. With how eachOfLimit is executed since the first iteration is synchronous, it executed the second iteration and so on so this causes multiple calls to `test.done()`